src/serial.rs

```diff
@@ -124,36 +172,40 @@ pub struct Serial<W: Write> {
     in_buffer: VecDeque<u8>,
 
     // Used for notifying the driver about some in/out events.
-    interrupt_evt: EventFd,
+    interrupt_evt: T,
     out: W,
 }
 
-impl<W: Write> Serial<W> {
+/// Errors encountered while handling serial console operations.
+#[derive(Debug)]
+pub enum Error<E> {
```

Comment:
I think this could benefit from being more explicitly parameterized as 
```rust
pub enum Error<T: Trigger> {
    Trigger(T::E)
```

Reply:
Looks like with this approach there's a problem when `T` doesn't implement `Debug` (which is the case for `EventFd`), i.e. `unwrap()` is not allowed for `std::result::Result<(), Error<T>>` (unless T implements `Debug`).
Let's keep it like it is (at least for now), if there are no other major concerns.